kong/config_service.go

```diff
@@ -0,0 +1,49 @@
+package kong
+
+import (
+	"context"
+	"fmt"
+	"io"
+)
+
+// AbstractConfigService handles Config in Kong.
+type AbstractConfigService interface {
+	ReloadDeclarativeConfig(ctx context.Context, config any, checkHash int) error
```

Comment:
I think it would make sense to have the comment documenting the method on the interface level as well.